packages/@react-spectrum/menu/src/SubMenuTrigger.tsx

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {classNames, useIsMobileDevice} from '@react-spectrum/utils';
+import {MenuContext, SubMenuTriggerContext, useMenuStateContext} from './context';
+import {Popover} from '@react-spectrum/overlays';
+import React, {Key, ReactElement, useRef} from 'react';
+import ReactDOM from 'react-dom';
+import styles from '@adobe/spectrum-css-temp/components/menu/vars.css';
+import {UNSTABLE_useSubMenuTrigger} from '@react-aria/menu';
+import {UNSTABLE_useSubMenuTriggerState} from '@react-stately/menu';
+
+// TODO: Add shouldFlip and closeOnSelect if we feel like those should be customizable for SubMenuTrigger
+// Other MenuTriggerProps like onOpenChange and positioning stuff have been removed as per discussion
+interface SubMenuTriggerProps {
+  /**
+   * The contents of the SubMenuTrigger - a Item and a Menu.
+   */
+  children: ReactElement[],
+  targetKey: Key
+}
+
+export interface SpectrumSubMenuTriggerProps extends Omit<SubMenuTriggerProps, 'targetKey'> {}
+
+// TODO: got rid of user provided ref support since it doesn't really make sense for submenus IMO
+function SubMenuTrigger(props: SubMenuTriggerProps) {
+  let triggerRef = useRef<HTMLDivElement>();
+  let menuRef = useRef<HTMLDivElement>();
+  let {
+    children,
+    targetKey
+  } = props;
+
+  let [menuTrigger, menu] = React.Children.toArray(children);
+  let {popoverContainerRef, trayContainerRef, menu: parentMenuRef, menuTreeState, state} = useMenuStateContext();
+  let triggerNode = state.collection.getItem(targetKey);
+  let subMenuTriggerState = UNSTABLE_useSubMenuTriggerState({triggerKey: targetKey}, {...menuTreeState, ...state});
```

Comment:
Note how I combine the TreeState (from Menu's call of `useTreeState`) and MenuTreeState (from root MenuTrigger's call of `useMenuState`). Not sure if we have this pattern elsewhere but I need access to the tree collection and the open/close submenu handlers